examples/application/main.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"../../iis"
+)
+
+func main() {
+	log.Printf("Starting test...")
+
+	testAppPath := "C:\\inetpub\\testing"
+	testAppContent := "<html><head><title>IIS Testing</title></head><body><h1>Hello World!</h1></body></html>"
+	appPoolName := "testing"
+	applicationName := "testing"
+	websiteName := "Default Web Site"
+	testUrl := "http://localhost/testing"
+
+	err := createTestDirectory(testAppPath, testAppContent)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Printf("Creating IIS Client")
+	client, err := iis.NewClient()
+	if err != nil {
+		log.Fatalf("Error creating IIS client: %s", err)
+	}
+
+	log.Printf("Creating IIS App Pool %q", appPoolName)
+	err = createAppPool(client, appPoolName)
+	if err != nil {
+		_ = os.RemoveAll(testAppPath)
+		log.Fatal(err)
+	}
```

Comment:
as the Create function returns an error - we should be able to assume this works (as otherwise an error will be returned), as such we should be able to make this just:

```suggestion
	err := client.AppPools.Create(appPoolName)
	if err != nil {
		return fmt.Errorf("Error creating IIS App Pool: %s", err)
	}
```